ring_doorbell/__init__.py

```diff
@@ -37,164 +22,19 @@ class Ring(object):
-
-    def _authenticate(self, attempts=RETRY_TOKEN, session=None, wait=1.0):
-        """Authenticate user against Ring API."""
-        url = API_URI + NEW_SESSION_ENDPOINT
-        loop = 0
-        # make a copy as we're mutating headers in the loop below
-        # which would cause issues with _get_oauth_token()
-        # which expects a non mutated HEADERS copy
-        modified_headers = HEADERS.copy()
-        while loop <= attempts:
-            modified_headers['Authorization'] = \
-                'Bearer {}'.format(self._get_oauth_token())
-            loop += 1
-
-            try:
-                if session is None:
-                    req = self.session.post((url),
-                                            data=POST_DATA,
-                                            headers=modified_headers,
-                                            timeout=self._timeout)
-                else:
-                    req = session
-            except requests.exceptions.RequestException as err_msg:
-                _LOGGER.error("Error!! %s", err_msg)
-                raise
-
-            if not req:
-                time.sleep(wait)  # add a pause or you'll get rate limited
-                continue
+        self._oauth = Auth()
+        self._oauth.fetch_token(username, password, auth_callback)
 
-            # if token is expired, refresh credentials and try again
-            if req.status_code == 200 or req.status_code == 201:
-
-                # the only way to get a JSON with token is via POST,
-                # so we need a special conditional for 201 code
-                if req.status_code == 201:
-                    data = req.json().get('profile')
-                    self.token = data.get('authentication_token')
-
-                self.is_connected = True
-                self.params = {'api_version': API_VERSION,
-                               'auth_token': self.token}
-
-                if self._persist_token and self._push_token_notify_url:
-                    url = API_URI + PERSIST_TOKEN_ENDPOINT
-                    PERSIST_TOKEN_DATA['auth_token'] = self.token
-                    PERSIST_TOKEN_DATA['device[push_notification_token]'] = \
-                        self._push_token_notify_url
-                    req = self.session.put((url), headers=modified_headers,
-                                           data=PERSIST_TOKEN_DATA,
-                                           timeout=self._timeout)
-
-                # update token if reuse_session is True
-                if self._reuse_session:
-                    self.cache['account'] = self.username
-                    self.cache['token'] = self.token
-                    self.cache['auth'] = self.auth
-                    _save_cache(self.cache, self.cache_file)
-
-                return True
-
-        self.is_connected = False
-        req.raise_for_status()
-        return True
+        self.cache = self._oauth.cache
```

Comment:
We shouldn't expose these anymore. It feels weird. Is this because other parts of the lib use this ?